Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1352 @@
+                            {
+                                var symbol = _symbolMapper.ConvertZerodhaSymbolToLeanSymbol(tick.InstrumentToken);
+
+                                var bestBidQuote = tick.Bids[0];
+                                var bestAskQuote = tick.Offers[0];
+
+                                var time = tick.Timestamp ?? DateTime.UtcNow;
+
+                                EmitQuoteTick(symbol, time, bestBidQuote.Price, bestBidQuote.Quantity, bestAskQuote.Price, bestAskQuote.Quantity);
+
+                                if (_lastTradeTickTime != time)
+                                {
+                                    EmitTradeTick(symbol, time, tick.LastPrice, tick.LastQuantity);
+                                    _lastTradeTickTime = time;
+                                }
+                            }
+                        }
+                    }
+                }
+                else if (e.MessageType == WebSocketMessageType.Text)
+                {
+                    string message = Encoding.UTF8.GetString(e.Data.Take(e.Count).ToArray());
+
+                    JObject messageDict = Utils.JsonDeserialize(message);
+                    if ((string)messageDict["type"] == "order")
+                    {
+                        OnOrderUpdate(new Messages.Order(messageDict["data"]));
+                    }
+                    else if ((string)messageDict["type"] == "error")
+                    {
+                        OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, -1, $"Zerodha WSS Error. Data: {e.Data} Exception: {messageDict["data"]}"));
+
+                    }
+                }
+                else if (e.MessageType == WebSocketMessageType.Close)
+                {
+                    WebSocket.Close();
+                }
+            }
+            catch (Exception exception)
+            {
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, -1, $"Parsing wss message failed. Data: {e.Data} Exception: {exception}"));
+                throw;
+            }
+        }
+
+        private void EmitTradeTick(Symbol symbol, DateTime time, decimal price, decimal amount)
+        {
+            //Log.Trace($"Trade: {time:O} {symbol} - Price:{price}, Quantity:{amount}");
+
+            var tick = new Tick(time, symbol, string.Empty, symbol.ID.Market, Math.Abs(amount), price);
+
+            _aggregator.Update(tick);
+        }
+
+        private void EmitQuoteTick(Symbol symbol, DateTime time, decimal bidPrice, decimal bidSize, decimal askPrice, decimal askSize)
+        {
+            if (bidPrice > 0 && askPrice > 0)
+            {
+                //Log.Trace($"Quote: {time:O} {symbol} - BidPrice:{bidPrice}, BidSize:{bidSize}, AskPrice:{askPrice}, AskSize:{askSize}");
+
+                var tick = new Tick(time, symbol, string.Empty, string.Empty, bidSize, bidPrice, askSize, askPrice);
```

Comment:
nit - should remove commented code here and above

`Tick.Time` is expected to be in exchange TZ by the Lean engine, here I think UTC is being sent in?